openmetadata-airflow-apis/openmetadata_managed_apis/operations/last_dag_logs.py

```diff
@@ -48,29 +58,55 @@ def last_dag_logs(dag_id: str) -> Response:
             f"Cannot find any task instance for the last DagRun of {dag_id}."
         )
 
-    response = {}
+    raw_logs_str = None
 
     for task_instance in task_instances:
 
-        # Pick up the _try_number, otherwise they are adding 1
-        try_number = task_instance._try_number  # pylint: disable=protected-access
+        # Only fetch the required logs
+        if task_instance.task_id == task_id:
+
+            # Pick up the _try_number, otherwise they are adding 1
+            try_number = task_instance._try_number  # pylint: disable=protected-access
 
-        task_log_reader = TaskLogReader()
-        if not task_log_reader.supports_read:
-            return ApiResponse.server_error(
-                f"Task Log Reader does not support read logs."
-            )
+            task_log_reader = TaskLogReader()
+            if not task_log_reader.supports_read:
+                return ApiResponse.server_error(
+                    "Task Log Reader does not support read logs."
+                )
 
-        logs = "\n".join(
-            list(
+            # Even when generating a ton of logs, we just get a single element.
+            # Same happens when trying to call task_log_reader.read_log_chunks
+            # We'll create our own chunk size and paginate based on that
+            raw_logs_str = "".join(list(
                 task_log_reader.read_log_stream(
                     ti=task_instance,
                     try_number=try_number,
                     metadata=LOG_METADATA,
                 )
-            )
+            ))
+
+    if not raw_logs_str:
+        return ApiResponse.bad_request(
+            f"Can't fetch logs for DAG {dag_id} and Task {task_id}."
         )
 
-        response[task_instance.task_id] = ResponseFormat.b64_gzip_compression(logs)
+    # Split the string in chunks of size without
+    # having to know the full length beforehand
+    log_chunks = [chunk for chunk in iter(partial(StringIO(raw_logs_str).read, CHUNK_SIZE), '')]
```

Comment:
Split the log stream with our own control of the `CHUNK_SIZE`. Airflow is not really returning chunks, but rather the whole log in one go